cpp/include/cudf/column/column.hpp

```diff
@@ -159,6 +159,13 @@ class column {
    */
   [[nodiscard]] size_type size() const noexcept { return _size; }
 
+  /**
+   * @brief Returns the number of data bytes
+   *
+   * @return The number of bytes
+   */
+  [[nodiscard]] std::size_t size_bytes() const noexcept;
```

Comment:
I think a fundamental invariant is that `column.size() * size_of(column.type() == column.size_bytes()`. However, given that `column::size()` returns `int`, then that invariant can be violated which seems _bad_. 

Are we planning to make `column::size()` also return `size_t`?

Reply:
We are not changing `size()` but rather only adding `size_bytes()`.

I think the invariant argument may break down with compounds types. We are only trying to solve for strings columns at this point.